src/test/java/systemtests/EditCommandSystemTest.java

```diff
@@ -85,20 +87,24 @@ public void edit() throws Exception {
         index = INDEX_SECOND_PERSON;
         command = EditCommand.COMMAND_WORD_ABBREVIATIONS.iterator().next() + " " + index.getOneBased() + NAME_DESC_BOB
                 + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB + TAG_DESC_FRIEND + TAG_DESC_HUSBAND;
-        assertCommandSuccess(command, index, BOB);
+        personToEdit = getModel().getLatestPersonList().get(index.getZeroBased());
+        editedPerson = new PersonBuilder().withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
+                .withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)
+                .build(personToEdit.getTags());
+        assertCommandSuccess(command, index, editedPerson);
 
         /* Case: edit some fields -> edited */
         index = INDEX_FIRST_PERSON;
         command = EditCommand.COMMAND_WORD + " " + index.getOneBased() + TAG_DESC_FRIEND;
-        ReadOnlyPerson personToEdit = getModel().getLatestPersonList().get(index.getZeroBased());
-        editedPerson = new PersonBuilder(personToEdit).withTags(VALID_TAG_FRIEND).build();
+        personToEdit = getModel().getLatestPersonList().get(index.getZeroBased());
+        editedPerson = new PersonBuilder(personToEdit).withTags(VALID_TAG_FRIEND).build(personToEdit.getTags());
         assertCommandSuccess(command, index, editedPerson);
 
-        /* Case: clear tags -> cleared */
+        /* Case: empty tag parameter -> no effect */
```

Comment:
Do you think it'll be better if we keep the functionality for clearing all tags?

Reply:
Perhaps. I'll try to work something in.

Reply:
@jo-lyn Nvm merge first. That one can do another time.